nengo/linear_system.py

```diff
@@ -0,0 +1,382 @@
+import numpy as np
+
+from nengo.base import Process
+from nengo.exceptions import ValidationError
+from nengo.params import BoolParam, EnumParam, NdarrayParam, Parameter
+from nengo.rc import rc
+from nengo._vendor.scipy.signal import (
+    cont2discrete,
+    ss2tf,
+    tf2ss,
+    tf2zpk,
+    zpk2ss,
+)
+
+
+def _as2d(M, last=True, dtype=np.float64):
+    if M is None:
+        return M
+
+    M = np.asarray(M, dtype=dtype)
+    if M.ndim < 2:
+        return M.reshape((1, -1)) if last else M.reshape((-1, 1))
+    else:
+        return M
+
+
+def _zeros_if_empty(M, shape, dtype=np.float64):
+    return np.zeros(shape, dtype=dtype) if M is None or M.size == 0 else M
+
+
+class LinearSystem(Process):
+    """General linear time-invariant (LTI) system.
+
+    Parameters
+    ----------
+    sys : list or tuple
+        A list or tuple representing a system in one of the following forms:
+
+            * length 2: (numerator, denominator) transfer function form [1]_
+            * length 3: (zero, pole, gain) zero-pole-gain form [2]_
+            * length 4: (A, B, C, D) state-space form [3]_
+
+        For more details on these forms, see `scipy.signal.lti`.
+
+    analog : boolean, optional
+        Whether ``sys`` is in analog (continuous-time) or discrete form.
+    method : string, optional
+        The method to use for discretization (if ``analog`` is True). See
+        `scipy.signal.cont2discrete` for information about the options.
+    x0 : array_like, optional
+        Initial values for the system state. The last dimension must equal the
+        ``state_size``.
+    default_dt : float, optional
+        If ``dt`` is not specified in `~.Process.run`, `~.Process.run_steps`,
+        `~.Process.ntrange`, or `~.Process.trange`, this will be used.
```

Comment:
I initially tried to make this reference `LinearSystem` rather than `Process`, but the docs build failed because it couldn't find a reference to those methods.